Rename the Delimiter

schlunz opened the pull request:
Dear @otoolep,

as discussed in the previous PRs,
this commit simply renames the existing `Delimiter` to be the `FallbackDelimiter` throughout the project.
This change is needed, in order to integrate a new delimiter (see #31). 

Sincerely, schlunz.

otoolep commented on input/delimiter/fallback_delimiter.go:
```diff
@@ -19,23 +19,23 @@ func init() {
-// A Delimiter detects when Syslog lines start.
-type Delimiter struct {
+// A FallbackDelimiter detects when Syslog lines start.
+type FallbackDelimiter struct {
```
Calling this a "fallback delimiter" isn't right. It's just a particular type of delimiter. The policy that the code will fall back to this one is made at a higher-level in the code.

Let's call it the "syslog delimiter" or something similar. Make sense?

schlunz commented:
@otoolep, after having some weird [CircleCI](https://circleci.com/gh/schlunz/ekanite/5) failure, I moved the delimiter back into the `input` package (and renamed it , to be the `SyslogDelimiter{}`).
